hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/server/TestSCMCertStore.java

```diff
@@ -0,0 +1,248 @@
+            CertificateStore.CertType.REVOKED_CERTS));
+
+    // CRL Info table should have a CRL with sequence id
+    assertNotNull(scmMetadataStore.getCRLInfoTable()
+        .get(sequenceId.get()));
+
+    // Check the sequence ID table for latest sequence id
+    assertEquals(INITIAL_SEQUENCE_ID + 1L, (long)
+        scmMetadataStore.getCRLSequenceIdTable().get(CRL_SEQUENCE_ID_KEY));
+
+    CRLInfo crlInfo =
+        scmMetadataStore.getCRLInfoTable().get(sequenceId.get());
+
+    Set<? extends X509CRLEntry> revokedCertificates =
+        crlInfo.getX509CRL().getRevokedCertificates();
+    assertEquals(1L, revokedCertificates.size());
+    assertEquals(x509Certificate.getSerialNumber(),
+        revokedCertificates.iterator().next().getSerialNumber());
+
+    // Now trying to revoke the already revoked certificate should result in
+    // a warning message and no-op. It should not create a new CRL.
+    sequenceId = scmCertStore.revokeCertificates(certs,
+        caCertificateHolder,
+        CRLReason.lookup(CRLReason.unspecified), crlApprover);
+
+    assertFalse(sequenceId.isPresent());
+
+    int size = 0;
+    TableIterator<Long, ? extends Table.KeyValue<Long, CRLInfo>> iter =
+        scmMetadataStore.getCRLInfoTable().iterator();
+
+    while(iter.hasNext()) {
+      size++;
+      iter.next();
+    }
+
+    assertEquals(1, size);
+
+    // Generate 3 more certificates and revoke 2 of them
+    List<BigInteger> newSerialIDs = new ArrayList<>();
+    for (int i = 0; i<3; i++) {
+      X509Certificate cert = generateX509Cert();
+      scmCertStore.storeValidCertificate(cert.getSerialNumber(), cert);
+      newSerialIDs.add(cert.getSerialNumber());
+    }
+
+    // Add the first 2 certificates to the revocation list
+    sequenceId = scmCertStore.revokeCertificates(newSerialIDs.subList(0, 2),
+        caCertificateHolder,
+        CRLReason.lookup(CRLReason.aACompromise), crlApprover);
+
+    // This should create a CRL with sequence id INITIAL_SEQUENCE_ID + 2
+    // And contain 2 certificates in it
+    assertTrue(sequenceId.isPresent());
+    assertEquals(INITIAL_SEQUENCE_ID + 2L, (long) sequenceId.get());
+
+    // Check the sequence ID table for latest sequence id
+    assertEquals(INITIAL_SEQUENCE_ID + 2L, (long)
+        scmMetadataStore.getCRLSequenceIdTable().get(CRL_SEQUENCE_ID_KEY));
+
+    CRLInfo newCrlInfo = scmMetadataStore.getCRLInfoTable()
+        .get(sequenceId.get());
+    revokedCertificates = newCrlInfo.getX509CRL().getRevokedCertificates();
+    assertEquals(2L, revokedCertificates.size());
+    assertNotNull(
+        revokedCertificates.stream().filter(c ->
+            c.getSerialNumber().equals(newSerialIDs.get(0)))
+            .findAny());
+
+    assertNotNull(
+        revokedCertificates.stream().filter(c ->
+            c.getSerialNumber().equals(newSerialIDs.get(1)))
+            .findAny());
+
+    // Valid certs table should have 1 cert
+    size = 0;
+    TableIterator<BigInteger,
```

Comment:
NIT: this can be a helper function like getTableSize() to avoid duplicated code.